best-practices/rspec.md

```diff
@@ -0,0 +1,30 @@
+## Rspec style guide
+
+- First #describe what you are doing
+- Then establish the #context
+- #it only expects one thing (within reason)
+- Prefer explicitness - #it #its #specify #subject are nice shortcuts but
+consider whether they are sacrificing readability
+- Having excessive setup (fixtures, database entities, doubles etc.) can cause false-positives. Thus, set up preconditions as close to the test as possible (usually within the context block) and tear them down afterwards. This is really important when working with time for example.
+- Unit tests are fast and effective to work with when doing TDD. These are isolated tests, typically
+for a single class, and should use stubs and spies via test doubles to replace anything that is
```

Comment:
I think I have the same thing - where the stub setup ends up four levels deep? But I think it's the test telling you you're breaking the law of demeter and you need to put a method on the first thing you call that goes and gets the thing deeper down that you want. No?